Current dev won't compile for NodeMCU board revision 0.9

When enabling DEVKIT_VERSION_0_9 in user_config.h, READLINE_INTERVAL is not defined and compilation crashes almost instantly.
In previous dev revisions, even with SDK 1.4, this was explicitelly defined, regardles of DEVKIT_VERSION and so compilation worked.
Further, with DEVKIT_VERSION_0_9 enabled, compilation crashes on node.c with following error:

```
node.c:246:6: error: conflicting types for 'key_long_press' [-Werror]
 void key_long_press(void *arg) {
      ^
node.c:183:7: note: previous implicit declaration of 'key_long_press' was here
       key_long_press(NULL);
       ^
node.c:258:6: error: conflicting types for 'key_short_press' [-Werror]
 void key_short_press(void *arg) {
      ^
node.c:187:7: note: previous implicit declaration of 'key_short_press' was here
       key_short_press(NULL);
       ^
cc1: all warnings being treated as errors
```

I tried rebuilding esp-open-sdk with crostools from https://github.com/pfalcon/esp-open-sdk.git (which worked before and is, as far as I know, suggested set for building the firmware) but it made no difference.

If DEVKIT_VERSION_0_9 is undefined, compilation works without problems.

It seems that somebody made a change without checking output on both board revisions and that rendered compilation for old board impossible.
Building is done on Fedora 21 and Ubuntu 12.04, no difference.

For some reason, frightanic's dev build still passes, don't know how if he also had his dev tools/SDK updated...?

--- a/app/include/user_config.h
+++ b/app/include/user_config.h
@@ -62,6 +62,7 @@
 #define LUA_OPTIMIZE_MEMORY         0
 #endif	/* LUA_OPTRAM */
 
+#define READLINE_INTERVAL 80
 #define LUA_TASK_PRIO USER_TASK_PRIO_0
 #define LUA_PROCESS_LINE_SIG 2
 

--- a/app/modules/node.c
+++ b/app/modules/node.c
@@ -152,6 +152,47 @@ static int key_press_count = 0;
 static bool key_short_pressed = false;
 static bool key_long_pressed = false;
 static os_timer_t keyled_timer;
+static int long_key_ref = LUA_NOREF;
+static int short_key_ref = LUA_NOREF;
+
+static void default_long_press(void *arg) {
+  if (led_high_count == 12 && led_low_count == 12) {
+    led_low_count = led_high_count = 6;
+  } else {
+    led_low_count = led_high_count = 12;
+  }
+  // led_high_count = 1000 / READLINE_INTERVAL;
+  // led_low_count = 1000 / READLINE_INTERVAL;
+  // NODE_DBG("default_long_press is called. hc: %d, lc: %d\n", led_high_count, led_low_count);
+}
+
+static void default_short_press(void *arg) {
+  system_restart();
+}
+
+static void key_long_press(void *arg) {
+  NODE_DBG("key_long_press is called.\n");
+  if (long_key_ref == LUA_NOREF) {
+    default_long_press(arg);
+    return;
+  }
+  if (!gL)
+    return;
+  lua_rawgeti(gL, LUA_REGISTRYINDEX, long_key_ref);
+  lua_call(gL, 0, 0);
+}
+
+static void key_short_press(void *arg) {
+  NODE_DBG("key_short_press is called.\n");
+  if (short_key_ref == LUA_NOREF) {
+    default_short_press(arg);
+    return;
+  }
+  if (!gL)
+    return;
+  lua_rawgeti(gL, LUA_REGISTRYINDEX, short_key_ref);
+  lua_call(gL, 0, 0);
+}
 
 static void update_key_led (void *p)
 {
@@ -223,48 +264,6 @@ static int node_led( lua_State* L )
   led_low_count = (uint32_t)low / READLINE_INTERVAL;
   prime_keyled_timer();
   return 0;
-}
-
-static int long_key_ref = LUA_NOREF;
-static int short_key_ref = LUA_NOREF;
-
-void default_long_press(void *arg) {
-  if (led_high_count == 12 && led_low_count == 12) {
-    led_low_count = led_high_count = 6;
-  } else {
-    led_low_count = led_high_count = 12;
-  }
-  // led_high_count = 1000 / READLINE_INTERVAL;
-  // led_low_count = 1000 / READLINE_INTERVAL;
-  // NODE_DBG("default_long_press is called. hc: %d, lc: %d\n", led_high_count, led_low_count);
-}
-
-void default_short_press(void *arg) {
-  system_restart();
-}
-
-void key_long_press(void *arg) {
-  NODE_DBG("key_long_press is called.\n");
-  if (long_key_ref == LUA_NOREF) {
-    default_long_press(arg);
-    return;
-  }
-  if (!gL)
-    return;
-  lua_rawgeti(gL, LUA_REGISTRYINDEX, long_key_ref);
-  lua_call(gL, 0, 0);
-}
-
-void key_short_press(void *arg) {
-  NODE_DBG("key_short_press is called.\n");
-  if (short_key_ref == LUA_NOREF) {
-    default_short_press(arg);
-    return;
-  }
-  if (!gL)
-    return;
-  lua_rawgeti(gL, LUA_REGISTRYINDEX, short_key_ref);
-  lua_call(gL, 0, 0);
 }
 
 // Lua: key(type, function)

